src/wallet/wallet.cpp

```diff
@@ -2421,103 +2421,84 @@ static void ApproximateBestSubset(const std::vector<CInputCoin>& vValue, const C
     }
 }
 
-bool CWallet::SelectCoinsMinConf(const CAmount& nTargetValue, const int nConfMine, const int nConfTheirs, const uint64_t nMaxAncestors, std::vector<COutput> vCoins,
```

Comment:
I am unsure yet, but I think you can make the diff way smaller by keeping the old signature of this method and building the `OutputGroup` at the scope of this function only.

Reply:
This method doesn't know whether you want single or grouped as it doesn't have coincontrol. Would have to add a bool for that at least. Not sure it'll become smaller though. It'll just move code around.

Reply:
So the way I did it with NBitcoin is to keep the concept of group internal to only the method which does coin packing.

If grouping is deactivated, I group by random, that way there is a single code path for everything.

I think it is a way smaller code change because all consumer of `SelectCoinsMinConf` do not have to change (you can make the `bool` default to `false` to keep original behavior)
The concept of group does not make lot's of sense anywhere outside of this method.

Reply:
I discussed with @NicolasDorier and we concluded that the current way this is done is okay (NBitcoin does it differently so it made sense there).